Tribler/Core/Modules/restapi/my_channel_endpoint.py

```diff
@@ -33,25 +49,43 @@ class MyChannelOverviewEndpoint(resource.Resource):
     }
     """
 
-    def __init__(self, session):
-        resource.Resource.__init__(self)
-        self.session = session
-        self.channel_db_handler = self.session.open_dbhandler(NTFY_CHANNELCAST)
-
-    def return_404(self, request):
-        """
-        Returns a 404 response code if your channel has not been created.
-        """
-        request.setResponseCode(404)
-        request.finish()
-
     def render_GET(self, request):
         my_channel_id = self.channel_db_handler.getMyChannelId()
         if my_channel_id is None:
-            self.return_404(request)
-            return server.NOT_DONE_YET
+            return MyChannelBaseEndpoint.return_404(request)
 
         my_channel = self.channel_db_handler.getChannel(my_channel_id)
         request.setHeader('Content-Type', 'text/json')
         return json.dumps({'overview': {'identifier': my_channel[1].encode('hex'), 'name': my_channel[2],
                                         'description': my_channel[3]}})
+
+
+class MyChannelTorrentsEndpoint(MyChannelBaseEndpoint):
+    """
+    This endpoint returns a 404 HTTP response if you have not created a channel (yet).
```

Comment:
I think the negative (404) case should go at the end, the main purpose of this class is not to return 404's (same with the previous one).